src/mtev_conf.c

```diff
@@ -3776,8 +3776,39 @@ static void safe_free_xpath(void *in) {
   if(gen && gen->xpath) xmlXPathFreeContext(gen->xpath);
   free(gen);
 }
+
+static mtev_hook_return_t
+mtev_conf_env_fixup(void *closure, mtev_conf_section_t section, const char *xpath,
+                    const char *nodepath, int set, char **value) {
+  (void)closure;
+  (void)section;
+  (void)xpath;
+  (void)nodepath;
+  if(set && *value && !strncmp(*value, "ENV:", 4)) {
+    char *fallback = *value + 4;
+    char *key = strchr(fallback, ':');
+    if(key){
+      key++;
+      if(*key == '{' && key[strlen(key)-1] == '}') {
+        key++;
+        char key_copy[128];
+        snprintf(key_copy, sizeof(key_copy), "%.*s", (int)strlen(key)-1, key);
+        char *replace = getenv(key_copy);
+        if(replace) replace = strdup(replace);
+        else replace = strndup(fallback, key - fallback - 1);
```

Comment:
```suggestion
        else replace = strndup(fallback, key - fallback - 2);
```

Reply:
key was already on the colon and also incremented twice after, so you need to subtract 2 not 1

Reply:
if the key was on the colon, we'd subtract 0... now that it has moved forward, we must subtract 1.

Reply:
it has moved forward twice

Reply:
32322:{C
^            ^
|.            0x1007 (key)
0x1000 (fallback)
0x1007 - 0x1000 - 1 = 6
but the length of 32322 should be 5

Reply:
You are correct... needed to remove the log as well

Reply:
ugh can't monospace font, the 0x1007 should be pointing to the first char of the env var, which is 2 past the colon